src/librustc_codegen_llvm/type_.rs

```diff
@@ -233,7 +234,26 @@ impl Type {
         }
     }
 
-    pub fn ptr_to(&self) -> &Type {
+    pub fn ptr_to(&self, cx: &CodegenCx<'ll, '_>) -> &Type {
+        if self.kind() == TypeKind::Function {
+            unsafe {
+                llvm::LLVMPointerType(self,
+                                      cx.data_layout().program_memory_address_space as c_uint)
```

Comment:
It is possible to "mis-use" this, I assume, for example via transmutes or something? Can we wind up with "valid Rust code" giving weird LLVM assertion failures or other errors?

Reply:
That's a good point. I'll see what happens if I transmute a few things to a function pointer.